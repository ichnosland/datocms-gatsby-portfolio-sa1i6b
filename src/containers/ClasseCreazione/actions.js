/*
 *
 * ClasseCreazione actions
 *
 */

import {
  CLASSE_CREAZIONE_SCUOLE_ATTIVE_FETCH,
  CLASSE_CREAZIONE_SCUOLE_ATTIVE_SET,
  CLASSE_CREAZIONE_GEO_PROVINCIA_FETCH,
  CLASSE_CREAZIONE_GEO_SET,
  CLASSE_CREAZIONE_GEO_COMUNI_FETCH,
  CLASSE_CREAZIONE_GEO_COMUNI_SET,
  CLASSE_CREAZIONE_DATA_SET,
  CLASSE_CREAZIONE_DATA_POST,
  CLASSE_CREAZIONE_RESET,
  CLASSE_CREAZIONE_SPINNER_SET,
  CLASSE_CREAZIONE_FEEDBACK_SET,
  CLASSE_CREAZIONE_FEEDBACK_RESET,
  CLASSE_CREAZIONE_GEO_SCUOLE_SET,
} from './constants';


export const classeCreazioneScuoleAttiveFetch = () => ({
  type: CLASSE_CREAZIONE_SCUOLE_ATTIVE_FETCH,
});

export const classeCreazioneScuoleAttiveSet = (payload) => ({
  type: CLASSE_CREAZIONE_SCUOLE_ATTIVE_SET,
  payload,
});

export const classeCreazioneGeoProvinciaFetch = () => ({
  type: CLASSE_CREAZIONE_GEO_PROVINCIA_FETCH,
});

export const classeCreazioneGeoSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_SET,
  payload,
});

export const classeCreazioneGeoComuneFetch = (sigla) => ({
  type: CLASSE_CREAZIONE_GEO_COMUNI_FETCH,
  sigla,
});

export const classeCreazioneGeoComuneSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_COMUNI_SET,
  payload,
});

export const classeCreazioneGeoScuoleSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_SCUOLE_SET,
  payload,
});

export const classeCreazioneDataSet = (payload) => ({
  type: CLASSE_CREAZIONE_DATA_SET,
  payload,
});

export const classeCreazioneDataPost = (payload, history) => ({
  type: CLASSE_CREAZIONE_DATA_POST,
  payload,
  history,
});

export const classeCreazioneReset = () => ({
  type: CLASSE_CREAZIONE_RESET,
});

export const classeCreazioneSpinnerSet = (enable) => ({
  type: CLASSE_CREAZIONE_SPINNER_SET,
  enable,
});

export const classeCreazioneFeedbackSet = (enable, tipologia, messaggio) => ({
  type: CLASSE_CREAZIONE_FEEDBACK_SET,
  enable,
  tipologia,
  messaggio,
});

export const classeCreazioneFeedbackReset = () => ({
  type: CLASSE_CREAZIONE_FEEDBACK_RESET,
});
