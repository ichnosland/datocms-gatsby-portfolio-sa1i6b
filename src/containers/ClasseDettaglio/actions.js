/*
 *
 * ClasseDettaglio actions
 *
 */

import {
  CLASSE_DETTAGLIO_ESPELLI,
  CLASSE_DETTAGLIO_RESET,
  CLASSE_DETTAGLIO_DATA_FETCH,
  CLASSE_DETTAGLIO_CORSO_FETCH,
  CLASSE_DETTAGLIO_CONTENUTO_SET,
  CLASSE_DETTAGLIO_DISPLAY_SET,
  CLASSE_DETTAGLIO_DISPLAY_SORT,
  CLASSE_DETTAGLIO_FEEDBACK_SET,
  CLASSE_DETTAGLIO_FEEDBACK_RESET,
  CLASSE_DETTAGLIO_SPINNER_SET,
} from './constants';

export const classeDettaglioEspelli = (payload) => ({
  type: CLASSE_DETTAGLIO_ESPELLI,
  payload,
});

export const classeDettaglioReset = () => ({
  type: CLASSE_DETTAGLIO_RESET,
});

export const classeDettaglioCorsoFetch = (payload) => ({
  type: CLASSE_DETTAGLIO_CORSO_FETCH,
  payload,
});

export const classeDettaglioDataFetch = (payload) => ({
  type: CLASSE_DETTAGLIO_DATA_FETCH,
  payload,
});

export const classeDettaglioContenutoSet = (payload) => ({
  type: CLASSE_DETTAGLIO_CONTENUTO_SET,
  payload,
});

export const classeDettaglioDisplaySort = (payloadData, sortingData) => ({
  type: CLASSE_DETTAGLIO_DISPLAY_SORT,
  payloadData,
  sortingData,
});

export const classeDettaglioDisplaySet = (payload) => ({
  type: CLASSE_DETTAGLIO_DISPLAY_SET,
  payload,
});

export const classeDettaglioFeedbackSet = (hasFeedback, tipologia, messaggio) => ({
  type: CLASSE_DETTAGLIO_FEEDBACK_SET,
  hasFeedback,
  tipologia,
  messaggio,
});

export const classeDettaglioFeedbackReset = () => ({
  type: CLASSE_DETTAGLIO_FEEDBACK_RESET,
});

export const classeDettaglioSpinnerSet = (enable) => ({
  type: CLASSE_DETTAGLIO_SPINNER_SET,
  enable,
});
