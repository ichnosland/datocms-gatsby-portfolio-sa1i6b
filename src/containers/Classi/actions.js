/*
 *
 * Classi actions
 *
 */

import {
  CLASSI_DATA_FETCH,
  CLASSI_DATA_SET,
  CLASSI_DATA_RESET,
  CLASSI_FEEDBACK_RESET,
  CLASSI_FEEDBACK_SET,
  CLASSI_SPINNER_SET,
} from './constants';

export const classiDataFetch = (idDisciplina) => ({
  type: CLASSI_DATA_FETCH,
  idDisciplina,
});

export const classiDataSet = (payload) => ({
  type: CLASSI_DATA_SET,
  payload,
});

export const classiDataReset = () => ({
  type: CLASSI_DATA_RESET,
});

export const classiFeedbackReset = () => ({
  type: CLASSI_FEEDBACK_RESET,
});

export const classiFeedbackSet = (enable = true, message = '', errorData = {}) => ({
  type: CLASSI_FEEDBACK_SET,
  enable,
  message,
  errorData,
});

export const classiSpinnerSet = (enable) => ({
  type: CLASSI_SPINNER_SET,
  enable,
});
