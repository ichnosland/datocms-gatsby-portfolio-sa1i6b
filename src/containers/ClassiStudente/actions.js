/*
 *
 * ClassiStudente actions
 *
 */

import {
  CLASSI_STUDENTE_FETCH,
  CLASSI_STUDENTE_SET,
  CLASSI_STUDENTE_RESET,
  CLASSI_STUDENTE_ISCRIVITI_POST,
  CLASSI_STUDENTE_SPINNER_SET,
  CLASSI_STUDENTE_FEEDBACK_SET,
  CLASSI_STUDENTE_FEEDBACK_RESET,
} from './constants';

export const classiStudenteFetch = (disciplinaId) => ({
  type: CLASSI_STUDENTE_FETCH,
  disciplinaId,
});

export const classiStudenteSet = (payload) => ({
  type: CLASSI_STUDENTE_SET,
  payload,
});

export const classiStudenteReset = () => ({
  type: CLASSI_STUDENTE_RESET,
});

export const classiStudenteIscrivitiPost = (codiceCorso, configuration) => ({
  type: CLASSI_STUDENTE_ISCRIVITI_POST,
  codiceCorso,
  configuration,
});

export const classiStudenteSpinnerSet = (enable) => ({
  type: CLASSI_STUDENTE_SPINNER_SET,
  enable,
});

export const classiStudenteFeedbackSet = (enable, tipologia, messaggio) => ({
  type: CLASSI_STUDENTE_FEEDBACK_SET,
  enable,
  tipologia,
  messaggio,
});

export const classiStudenteFeedbackReset = () => ({
  type: CLASSI_STUDENTE_FEEDBACK_RESET,
});
