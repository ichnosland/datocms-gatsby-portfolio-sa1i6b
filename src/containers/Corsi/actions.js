/*
 *
 * Corsi actions
 *
 */

import {
  CORSI_RESET,
  CORSI_ERROR_SET,
  CORSI_ERROR_RESET,
  CORSI_SPINNER_SET,
  CORSI_LIST_INITIALIZE,
  CORSI_LIST_DOCENTE_FETCH,
  CORSI_LIST_STUDENTE_FETCH,
  CORSI_LIST_SET,
  CORSI_CORSO_DETAIL_DOCENTE_TRIGGER,
  CORSI_CORSO_DETAIL_DOCENTE_SET,
} from './constants';

export const corsiReset = () => ({
  type: CORSI_RESET,
});

export const corsiSpinnerSet = (enable) => ({
  type: CORSI_SPINNER_SET,
  enable,
});

export const corsiErrorSet = (enable = true, message = '', errorData = {}) => ({
  type: CORSI_ERROR_SET,
  enable,
  message,
  errorData,
});

export const corsiErrorReset = () => ({
  type: CORSI_ERROR_RESET,
});

export const corsiListInitialize = (configuration, isDocente, userId) => ({
  type: CORSI_LIST_INITIALIZE,
  configuration,
  isDocente,
  userId,
});

export const corsiListDocenteFetch = (idDisciplina) => ({
  type: CORSI_LIST_DOCENTE_FETCH,
  idDisciplina,
});

export const corsiListStudenteFetch = (idDisciplina) => ({
  type: CORSI_LIST_STUDENTE_FETCH,
  idDisciplina,
});

export const corsiListSet = (payload) => ({
  type: CORSI_LIST_SET,
  payload,
});

export const corsiCorsoSelectedDocenteTrigger = (idCorso, configuration, isDocente, userId) => ({
  type: CORSI_CORSO_DETAIL_DOCENTE_TRIGGER,
  idCorso,
  configuration,
  isDocente,
  userId,
});

export const corsiCorsoDetailDocenteSet = (payload) => ({
  type: CORSI_CORSO_DETAIL_DOCENTE_SET,
  payload,
});
