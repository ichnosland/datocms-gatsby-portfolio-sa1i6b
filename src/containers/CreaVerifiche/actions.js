/*
 *
 * CreaVerifiche actions
 *
 */

import {
  CREA_VERIFICHE_FETCH,
  CREA_VERIFICHE_FETCH_ERROR,
  CREA_VERIFICHE_PREVIEW_ESERCIZIO_FETCH,
  CREA_VERIFICHE_PREVIEW_ESERCIZIO_FETCH_ERROR,
  CREA_VERIFICHE_SET,
  CREA_VERIFICHE_SPINNER_SET,
  CREA_VERIFICHE_POST,
  CREA_VERIFICHE_POST_ERROR,
  CREA_VERIFICHE_VERIFICA_SET,
  CREA_VERIFICHE_VERIFICA_ESERCIZIO_ADD,
  CREA_VERIFICHE_VERIFICA_ESERCIZIO_REMOVE,
  CREA_VERIFICHE_VERIFICA_RIEPILOGO_VIEW,
} from './constants';

export const creaverificheVerificaPostError = (messaggio = 'Impossibile inviare i dati') => ({
  type: CREA_VERIFICHE_POST_ERROR,
  messaggio,
});

export const creaverificheVerificaPost = (payload) => ({
  type: CREA_VERIFICHE_POST,
  payload,
});

export const creaverificheVerificaEserciziFetch = (prerequisito) => ({
  type: CREA_VERIFICHE_FETCH,
  prerequisito,
});

export const creaverificheVerificaEserciziFetchError = (messaggio = 'Impossibile caricare l\'elenco degli esercizi disponibili') => ({
  type: CREA_VERIFICHE_FETCH_ERROR,
  messaggio,
});

export const creaverificheVerificaEsercizioPreviewFetch = (esercizioId) => ({
  type: CREA_VERIFICHE_PREVIEW_ESERCIZIO_FETCH,
  esercizioId,
});

export const creaverificheVerificaEsercizioPreviewFetchError = (
  messaggio = 'Impossibile caricare l\'elenco degli step per questo esercizio'
) => ({
  type: CREA_VERIFICHE_PREVIEW_ESERCIZIO_FETCH_ERROR,
  messaggio,
});

export const creaverificheVerificaEserciziSet = (payload) => ({
  type: CREA_VERIFICHE_SET,
  payload,
});

export const creaverificheVerificaEserciziSpinnerSet = (enabled) => ({
  type: CREA_VERIFICHE_SPINNER_SET,
  enabled,
});

export const creaverificheVerificaSet = (payload) => ({
  type: CREA_VERIFICHE_VERIFICA_SET,
  payload,
});

export const creaverificheVerificaEsercizioAdd = (step) => ({
  type: CREA_VERIFICHE_VERIFICA_ESERCIZIO_ADD,
  step,
});

export const creaverificheVerificaEsercizioRemove = (step) => ({
  type: CREA_VERIFICHE_VERIFICA_ESERCIZIO_REMOVE,
  step,
});

export const creaverificheVerificaRiepilogoView = (enabled) => ({
  type: CREA_VERIFICHE_VERIFICA_RIEPILOGO_VIEW,
  enabled,
});
