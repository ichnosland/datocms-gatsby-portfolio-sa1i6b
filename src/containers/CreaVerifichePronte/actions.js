/*
 *
 * CreaVerifichePronte actions
 *
 */

import {
  CREA_VERIFICHE_PRONTE_FETCH,
  CREA_VERIFICHE_PRONTE_FETCH_ERROR,
  CREA_VERIFICHE_PRONTE_SET,
  CREA_VERIFICHE_PRONTE_SPINNER_SET,
  CREA_VERIFICHE_PRONTE_FETCH_ESERCIZI,
} from './constants';

export const creaverificheVerifichePronteFetch = (disciplinaId) => ({
  type: CREA_VERIFICHE_PRONTE_FETCH,
  disciplinaId,
});

export const creaverificheVerifichePronteFetchError = (messaggio = 'Impossibile caricare l\'elenco delle verifiche pronte') => ({
  type: CREA_VERIFICHE_PRONTE_FETCH_ERROR,
  messaggio,
});

export const creaverificheVerifichePronteSet = (payload) => ({
  type: CREA_VERIFICHE_PRONTE_SET,
  payload,
});

export const creaverificheVerifichePronteSpinnerSet = (enabled) => ({
  type: CREA_VERIFICHE_PRONTE_SPINNER_SET,
  enabled,
});

export const creaverificheVerifichePronteEserciziFetch = (payload) => ({
  type: CREA_VERIFICHE_PRONTE_FETCH_ESERCIZI,
  payload,
});
