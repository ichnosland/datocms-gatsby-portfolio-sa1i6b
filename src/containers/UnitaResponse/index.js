/**
 *
 * UnitaResponse
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';

import { unitaReset } from 'containers/Unita/actions';
import ResponseView from 'components/ResponseView';
import goodBg from './images/good.gif';


export class UnitaResponseView extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  componentDidMount() {
    const {
      contenutoUnita: { isLoaded, id, consegnata },
      history,
      configuration: { homePage },
    } = this.props;

    if (!isLoaded || !consegnata) {
      history.push(id > 0 ? `/unita-preview/${id}` : homePage);
    }
  }

  render() {
    const {
      history,
      configuration,
      onPurgeUnita,
      contenutoUnita: { id, lezioni: { completate, totali } },
      userAppData: { docente },
    } = this.props;

    return (
      <ResponseView
        votoApprossimato={10}
        goodSrc={goodBg}
        nascondiVoto
        resetFunction={() => {
          onPurgeUnita();
          history.push(`/unita-preview/${id}`);
        }}
        titolo="Hai completato la lezione"
        steps={!docente}
        lezioniCompletate={completate}
        lezioniTotali={totali}
        product={configuration.product}
      />
    );
  }
}

UnitaResponseView.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }),
  configuration: PropTypes.shape({
    homePage: PropTypes.string.isRequired,
    product: PropTypes.string.isRequired,
  }).isRequired,
  contenutoUnita: PropTypes.shape({
    consegnata: PropTypes.bool.isRequired,
    isLoaded: PropTypes.bool.isRequired,
    id: PropTypes.number.isRequired,
    lezioni: PropTypes.shape({
      totali: PropTypes.number.isRequired,
      completate: PropTypes.number.isRequired,
    }).isRequired,
  }).isRequired,
  onPurgeUnita: PropTypes.func.isRequired,
  userAppData: PropTypes.shape({
    docente: PropTypes.bool.isRequired,
  }).isRequired,
};


const mapDispatchToProps = (dispatch) => ({
  onPurgeUnita: () => {
    dispatch(unitaReset());
  },
});

const mapStateToProps = (state) => ({
  configuration: state.get('configuration').toJS(),
  userAppData: state.get('user').toJS().appData,
  spinner: state.get('unita').toJS().spinner,
  feedback: state.get('unita').toJS().feedback,
  contenutoUnita: state.get('unita').toJS().contenuto,
  step: state.get('unita').toJS().step,
  risposta: state.get('unita').toJS().risposta,
});

const withConnect = connect(mapStateToProps, mapDispatchToProps);

const connectData = compose(
  withConnect,
)(UnitaResponseView);

export default connectData;
