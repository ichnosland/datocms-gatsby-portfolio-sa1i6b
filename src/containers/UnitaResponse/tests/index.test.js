import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'configureStore';
import { compose } from 'redux';
import { connect } from 'react-redux';

import injectReducer from 'utils/injectReducer';

import reducerCurryer from 'common/applications/reducer';
import { Button } from 'components/Button';
import userReducer, { defaultAppData } from 'containers/User/reducer';
import { unitaReset } from 'containers/Unita/actions';
import ResponseView from 'components/ResponseView';
import unitaReducer, {
  defaultUnitaFeedback,
  defaultUnitaContenuto,
  defaultUnitaRisposta,
  defaultUnitaStep,
} from 'containers/Unita/reducer';
import UnitaResponse, { UnitaResponseView } from '../index';


const mockConfiguration = {
  product: 'product',
  hasPremium: true,
  disciplinaId: 123,
  homePage: '/homepage',
};

const baseProps = {
  userAppData: {
    docente: false,
  },
  configuration: mockConfiguration,
  contenutoUnita: {
    isLoaded: true,
    id: 3,
    consegnata: true,
    lezioni: {
      completate: 2,
      totali: 7,
    },
  },
  history: {
    push: () => { },
  },
  onPurgeUnita: () => { },
};

describe('<UnitaResponseView />', () => {
  it('visualizza il riepilogo con le informazioni di base', () => {
    const renderedComponent = shallow(
      <UnitaResponseView {...baseProps} />
    );
    expect(renderedComponent.dive().find(Button).length).toBe(1);
    expect(renderedComponent.find(ResponseView).props()).toEqual({
      votoApprossimato: 10,
      goodSrc: 'IMAGE_MOCK',
      lezioniCompletate: 2,
      lezioniTotali: 7,
      titolo: 'Hai completato la lezione',
      steps: true,
      resetFunction: expect.any(Function),
      nascondiVoto: true,
      product: 'product',
    });
    expect(renderedComponent).toMatchSnapshot();
  });

  it('visualizza il riepilogo con le informazioni di base se docente', () => {
    const props = {
      ...baseProps,
      userAppData: {
        docente: true,
      },
    };
    const renderedComponent = shallow(
      <UnitaResponseView {...props} />
    );
    expect(renderedComponent.dive().find(Button).length).toBe(1);
    expect(renderedComponent.find(ResponseView).props()).toEqual({
      votoApprossimato: 10,
      goodSrc: 'IMAGE_MOCK',
      lezioniCompletate: 2,
      lezioniTotali: 7,
      titolo: 'Hai completato la lezione',
      steps: false,
      resetFunction: expect.any(Function),
      nascondiVoto: true,
      product: 'product',
    });
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render a ResponseWrap with an img', () => {
    const renderedComponent = shallow(
      <UnitaResponseView {...baseProps} />
    );
    expect(renderedComponent.find(ResponseView).length).toBe(1);
    expect(renderedComponent.dive().find('img').length).toBe(1);
  });

  it('il click al button deve chiamare onPurgeUnita', () => {
    const props = {
      ...baseProps,
      onPurgeUnita: jest.fn(),
    };
    const renderedComponent = shallow(
      <UnitaResponseView {...props} />
    );
    expect(renderedComponent.dive().find(Button).length).toBe(1);
    expect(props.onPurgeUnita).not.toHaveBeenCalled();
    renderedComponent.dive().find(Button).simulate('click');
    expect(props.onPurgeUnita).toHaveBeenCalled();
  });

  it('fa un push di history se isLoaded = false e contenutoUnita.id > 0', () => {
    const props = {
      ...baseProps,
      contenutoUnita: {
        ...baseProps.contenutoUnita,
        isLoaded: false,
        id: 3,
      },
      history: {
        push: jest.fn(),
      },
    };
    expect(props.history.push).not.toHaveBeenCalled();
    shallow(<UnitaResponseView {...props} />);
    expect(props.history.push).toHaveBeenCalledWith('/unita-preview/3');
  });

  it('fa un push di history se consegnata = false se contenutoUnita.id > 0', () => {
    const props = {
      ...baseProps,
      contenutoUnita: {
        ...baseProps.contenutoUnita,
        consegnata: false,
        id: 3,
      },
      history: {
        push: jest.fn(),
      },
    };
    expect(props.history.push).not.toHaveBeenCalled();
    shallow(<UnitaResponseView {...props} />);
    expect(props.history.push).toHaveBeenCalledWith('/unita-preview/3');
  });

  it('fa un push di history se consegnata = false se contenutoUnita.id == 0', () => {
    const props = {
      ...baseProps,
      contenutoUnita: {
        ...baseProps.contenutoUnita,
        consegnata: false,
        id: 0,
      },
      history: {
        push: jest.fn(),
      },
    };
    expect(props.history.push).not.toHaveBeenCalled();
    shallow(<UnitaResponseView {...props} />);
    expect(props.history.push).toHaveBeenCalledWith(mockConfiguration.homePage);
  });
});

describe('<UnitaResponse />', () => {
  const withReducerConfiguration = injectReducer({ key: 'configuration', reducer: reducerCurryer(mockConfiguration) });
  const withReducerUser = injectReducer({ key: 'user', reducer: userReducer });
  const withReducerUnita = injectReducer({ key: 'unita', reducer: unitaReducer });

  const ConfigurationWrapper = compose(
    withReducerConfiguration,
    withReducerUser,
    withReducerUnita,
    connect()
  )(UnitaResponse);

  it('le props iniziali devono essere settate correttamente', () => {
    const store = configureStore({}, {});
    const renderedComponent = shallow(
      <ConfigurationWrapper
        history={{ push: () => { } }}
      />, {
        context: { store },
      }
    ).dive().dive().dive().dive().dive();

    const instance = renderedComponent.instance();
    const receivedProps = instance.props;
    expect(receivedProps.configuration).toEqual(mockConfiguration);
    expect(receivedProps.userAppData).toEqual(defaultAppData.toJS());
    expect(receivedProps.step).toEqual(defaultUnitaStep.toJS());
    expect(receivedProps.spinner).toBe(true);
    expect(receivedProps.feedback).toEqual(defaultUnitaFeedback.toJS());
    expect(receivedProps.risposta).toEqual(defaultUnitaRisposta.toJS());
    expect(receivedProps.contenutoUnita).toEqual(defaultUnitaContenuto.toJS());
  });

  it('la prop onPurgeUnita deve fare il dispatch di unitaReset', () => {
    const store = configureStore({}, {});
    store.dispatch = jest.fn();
    const renderedComponent = shallow(
      <ConfigurationWrapper
        history={{ push: () => { } }}
      />, {
        context: { store },
      }
    ).dive().dive().dive().dive().dive();

    const instance = renderedComponent.instance();
    const receivedProps = instance.props;
    expect(store.dispatch).not.toHaveBeenCalled();
    receivedProps.onPurgeUnita();
    expect(store.dispatch).toHaveBeenCalledWith(unitaReset());
  });
});
