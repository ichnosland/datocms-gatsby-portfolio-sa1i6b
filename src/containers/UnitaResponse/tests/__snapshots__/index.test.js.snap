// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnitaResponseView /> visualizza il riepilogo con le informazioni di base 1`] = `
<ResponseView
  goodSrc="IMAGE_MOCK"
  lezioniCompletate={2}
  lezioniTotali={7}
  nascondiVoto={true}
  product="product"
  resetFunction={[Function]}
  steps={true}
  titolo="Hai completato la lezione"
  votoApprossimato={10}
/>
`;

exports[`<UnitaResponseView /> visualizza il riepilogo con le informazioni di base se docente 1`] = `
<ResponseView
  goodSrc="IMAGE_MOCK"
  lezioniCompletate={2}
  lezioniTotali={7}
  nascondiVoto={true}
  product="product"
  resetFunction={[Function]}
  steps={false}
  titolo="Hai completato la lezione"
  votoApprossimato={10}
/>
`;
